imdclient/utils.py

```diff
@@ -43,24 +43,6 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # always propagate exceptions forward
         return False
 
-# NOTE: think of other edge cases as well- should be robust
-def parse_host_port(filename):
```

Comment:
Let's keep this function in imdclient because it's here where we define the "imd" protocol, not in MDA.

Just add an explicit test for it where you check that it can correctly parse a typical imd string and correctly raises it's ValueErrors.